### PR TITLE
Ecf 89823022 available content modules

### PR DESCRIPTION
**Pull request by ericcf**

### CODE REVIEW

File: app/models/tool_nav_item.rb [hunk 1]
```diff
@@ -34,33 +34,22 @@ def has_subnav?
   end
 
   def module_nav_items
-    ModuleNavItem.for_content_modules(@participant, available_content_modules)
+    available_content_modules
```
Comment (gwensmu) [inline app/models/tool_nav_item.rb]: so readable

File: app/models/tool_nav_item.rb [hunk 2]
```diff
@@ -34,33 +34,22 @@ def has_subnav?
...
 
   def any_incomplete_tasks_today?
-    content_module_ids = available_content_modules.where(is_viz: false)
-                         .map(&:id)
-
-    @participant.active_membership
-      .incomplete_tasks_today
-      .for_content_module_ids(content_module_ids)
-      .exists?
+    available_content_modules.is_not_viz.is_not_completed.exists?
```
Comment (gwensmu) [inline app/models/tool_nav_item.rb]: i could weep!

File: app/views/think_feel_do_engine/participants/content_modules/index.html.erb [hunk 3]
```diff
@@ -44,25 +43,20 @@
...
+        .new(self, nil, nil, current_participant),
+        navigator_location_path(
+          module_id: content_module.id
         )
-        .html_safe 
-        # render the viz provider
-    %>
-  <% end %>
+      )
+      .html_safe
```
Comment (gwensmu) [inline app/views/think_feel_do_engine/participants/content_modules/index.html.erb]: content_tag instead of html_safe?

Comment (ericcf) [inline app/views/think_feel_do_engine/participants/content_modules/index.html.erb]: I agree. I'll add a new commit.

Comment (ericcf) [inline app/views/think_feel_do_engine/participants/content_modules/index.html.erb]: Actually, this is a bigger refactor than I thought. Not sure it's feasible right now without a pretty big rewrite.

Comment (gwensmu) [inline app/views/think_feel_do_engine/participants/content_modules/index.html.erb]: ok. the content module titles aren't created via web anyway, they are imported direct to database, so this isn't as big of an issue as other uses, right?

Comment (ericcf) [inline app/views/think_feel_do_engine/participants/content_modules/index.html.erb]: It's actually calling the `render_current` method on a `ContentProvider` subclass instance, e.g. https://github.com/cbitstech/think_feel_do_engine/blob/ecf_89823022_available_content_modules/app/models/content_providers/activity_index_provider.rb#L9

**Comment by gwensmu** [comment]
🌅

**Pull request merged**